### DISCUSSION TITLE
Missing models

### DISCUSSION THREAD

**Discussion by quesnel** [vle-forge/irritator#41] [Ideas]
- `switch`
- `and`, `or`, `not`, `nand`, `nor`.

**Comment by quesnel** [comment]
We can split models into continuous (all current irritator model) and agent as in [Anylogic project](https://anylogic.help/library-reference-guides/process-modeling-library/delay.html).
- Delay (queue and dynamic queue in irritator)
- SelectOutput
- SelectOutput5
- Match

@duboz  Any ideas about missing continuous models?

**Reply by duboz** [reply]
We do not have partial differential equations systems. It can be useful for spatially explicit continuous systems. Most of their numerical solver schema are discrete time. Maybe not a priority for now as we do not currently work on this type of model.

**Reply by duboz** [reply]
OK, maybe change the "agent" for "object".

**Comment by quesnel** [comment]
@ovese can you develop a new model (as `irt::none`, `irt::counter` etc.). This model `irt::filter` as an input port and an output port. If the value (`irt::message`) receives on input port is lower than `up-threshold` and upper than `bottom-threshold` then the message is immediately sent otherwise model does nothing and the message is forgotten.

A priori I will need three patches:
1. Introduces the new model (without dynamics, just one input and one output port and a sigma equal to infinity). This patch modify `core.hpp`, `io.hpp` and  `node-editor`.
2. Add the behavior to the model (in `core.hpp`) with hard coded `up-threshold` and `bottom-threshold` value. Update the unit test to test your model.
3. Convert `up-threshold` and `bottom-threshold` into parameters and update `node-editor`.

@duboz Is it good for you?

**Reply by duboz** [reply]
The filter model you propose is a controller checking if a value belong to an interval. The specification is OK for me.

The "cross" model is not available anymore ? What is often necessary is to fix one single threshold and detect if it is up or down crossed.

**Comment by ovese** [comment]
Merci @duboz pour des prochaine taches. je vais commencer a regarde.

Le ven. 18 juin 2021 à 10:16, Gauthier Quesnel ***@***.***> a
écrit :

> @ovese <https://github.com/ovese> can you develop a new model (as
> irt::none, irt::counter etc.). This model irt::filter as an input port
> and an output port. If the value (irt::message) receives on input port is
> lower than up-threshold and upper than bottom-threshold then the message
> is immediately sent otherwise model does nothing and the message is
> forgotten.
>
> A priori I will need three patches:
>
>    1. Introduces the new model (without dynamics, just one input and one
>    output port and a sigma equal to infinity). This patch modify core.hpp,
>    io.hpp and node-editor.
>    2. Add the behavior to the model (in core.hpp) with hard coded
>    up-threshold and bottom-threshold value. Update the unit test to test
>    your model.
>    3. Convert up-threshold and bottom-threshold into parameters and
>    update node-editor.
>
> @duboz <https://github.com/duboz> Is it good for you?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vle-forge/irritator/discussions/41#discussioncomment-886997>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOUADDSTOXZ4QG5O6TSK2R3TTMFF3ANCNFSM46KBUWMA>
> .
>


-- 
Ovie Eseoghene(PhD.)
*Research Scientist (Systems Dynamics & Automatic Control)*
*NASRDA, Nigeria*
*PhD. Ahmadu Bello University Zaria, Nigeria, **MSc. INSA de Lyon, France
& **B.Eng. University of Benin, Nigeria*
*Cell:+234(0)8066765589*

**Comment by quesnel** [comment]
On lun. 21 juin 21, Raphaël Duboz wrote:
> The filter model you propose is a controller checking if a value
> belong to an interval. The specification is OK for me.

Ok.

> The "cross" model is not available anymore ? What is often necessary
> is to fix one single threshold and detect if it is up or down crossed.

The cross model is still available.
